### PR TITLE
Add dataset, module task, and demo of text-matching

### PR DESCRIPTION
**Pull request by KPatr1ck**

### CODE REVIEW

File: modules/text/language_model/ernie_tiny/module.py [hunk 1]
```diff
@@ -71,6 +71,12 @@ def __init__(
             self.metric = ChunkEvaluator(
                 label_list=[self.label_map[i] for i in sorted(self.label_map.keys())]
             )
+        elif task == 'text-matching':
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py:74]: 组网实现不是Sentence-Transformer

参考：https://github.com/PaddlePaddle/PaddleNLP/tree/develop/examples/text_matching/sentence_transformers

File: paddlehub/datasets/base_nlp_dataset.py [hunk 2]
```diff
@@ -469,3 +469,127 @@ def __getitem__(self, idx):
...
+        Converts all examples to records which the model needs.
+        Args:
+            examples(obj:`List[InputExample]`): All data examples returned by _read_file.
+        Returns:
+            records(:obj:`List[dict]`): All records which the model needs.
+        """
+        records = []
+        for example in examples:
+            if isinstance(self.tokenizer, PretrainedTokenizer):
+                if Version(paddlenlp.__version__) <= Version('2.0.0rc2'):
```
Comment (Steffy-zxf) [inline paddlehub/datasets/base_nlp_dataset.py]: 无须兼容PaddleNLP 2.0.0rc2 以下版本，匹配任务可以直接考虑paddlenlp 2.0.0rc6以上版本

File: paddlehub/datasets/base_nlp_dataset.py [hunk 3]
```diff
@@ -469,3 +469,127 @@ def __getitem__(self, idx):
...
+                record['label'] = self.label_map[example.label]
+            records.append(record)
+        return records
+
+    def __getitem__(self, idx):
+        record = self.records[idx]
+        input_ids = np.array(record['text_a']['input_ids'])
+        input_ids2 = np.array(record['text_b']['input_ids'])
+        if isinstance(self.tokenizer, PretrainedTokenizer):
+            if Version(paddlenlp.__version__) >= Version('2.0.0rc5'):
```
Comment (Steffy-zxf) [inline paddlehub/datasets/base_nlp_dataset.py]: 无须兼容PaddleNLP 2.0.0rc2 以下版本，匹配任务可以直接考虑paddlenlp 2.0.0rc6以上版本

File: paddlehub/module/nlp_module.py [hunk 4]
```diff
@@ -404,63 +404,80 @@ class TransformerModule(RunModule, TextServing):
     _tasks_supported = [
         'seq-cls',
         'token-cls',
+        'text-matching',
     ]
 
-    def _convert_text_to_input(self, tokenizer, text: List[str], max_seq_len: int, split_char: str):
+    def _convert_text_to_input(self, tokenizer, texts: List[str], max_seq_len: int, split_char: str):
```
Comment (Steffy-zxf) [inline paddlehub/module/nlp_module.py:410]: 建议不要轻易变动接口参数名

File: paddlehub/module/nlp_module.py [hunk 5]
```diff
@@ -404,63 +404,80 @@ class TransformerModule(RunModule, TextServing):
...
+        encoded_inputs = []
+        if self.task == 'text-matching':
+            if len(texts) != 2:
+                raise RuntimeError(
+                    'The input texts must have two sequences, but got %d. Please check your inputs.' % len(texts))
+
             if Version(paddlenlp.__version__) <= Version('2.0.0rc2'):
-                encoded_inputs = tokenizer.encode(
-                    text[0], text_pair=text[1], max_seq_len=max_seq_len, pad_to_max_seq_len=pad_to_max_seq_len)
+                encoded_inputs.append(tokenizer.encode(
```
Comment (Steffy-zxf) [inline paddlehub/module/nlp_module.py]: 无须兼容PaddleNLP 2.0.0rc2 以下版本，可以直接考虑paddlenlp 2.0.0rc6以上版本

File: paddlehub/module/nlp_module.py [hunk 6]
```diff
@@ -404,63 +404,80 @@ class TransformerModule(RunModule, TextServing):
...
-                    return_length=True)
+                encoded_inputs.append(tokenizer(text=texts[0], text_pair=None, max_seq_len=max_seq_len, \
+                        pad_to_max_seq_len=True, is_split_into_words=is_split_into_words, return_length=True))
+                encoded_inputs.append(tokenizer(text=texts[1], text_pair=None, max_seq_len=max_seq_len, \
+                        pad_to_max_seq_len=True, is_split_into_words=is_split_into_words, return_length=True))
         else:
-            raise RuntimeError(
-                'The input text must have one or two sequence, but got %d. Please check your inputs.' % len(text))
+            if len(texts) == 1:
+                if Version(paddlenlp.__version__) <= Version('2.0.0rc2'):
```
Comment (Steffy-zxf) [inline paddlehub/module/nlp_module.py:430]: 无须兼容PaddleNLP 2.0.0rc2 以下版本，可以直接考虑paddlenlp 2.0.0rc6以上版本

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]
还缺少文本匹配demo 脚本运行示例

File: modules/text/language_model/ernie_tiny/module.py [hunk 7]
```diff
@@ -84,8 +90,11 @@ def __init__(
             self.set_state_dict(state_dict)
             logger.info('Loaded parameters from %s' % os.path.abspath(load_checkpoint))
 
-    def forward(self, input_ids, token_type_ids=None, position_ids=None, attention_mask=None, seq_lengths=None, labels=None):
+    def forward(self, input_ids, token_type_ids=None, input_ids2=None, token_type_ids2=None, position_ids=None, attention_mask=None, seq_lengths=None, labels=None):
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py]: 建议保留原有`input_ids`, `token_type_ids`, `position_ids`, `attention_mask`, `seq_lengths`, `labels` 的参数，增加
`query_input_ids`, `query_token_type_ids`, `query_position_ids`, `query_attention_mask`, `query_seq_lengths`, 
`title_input_ids`, `title_token_type_ids`, `title_position_ids`, `title_attention_mask`, `title_seq_lengths` 
参数

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]

File: modules/text/language_model/ernie_tiny/module.py (hunk 1 above)
Comment (KPatr1ck) [reply modules/text/language_model/ernie_tiny/module.py:74]: 之前用的Sentence-Transformer中的CLS-token的方式，现已改为Mean Pooling，同步paddlenlp

File: paddlehub/datasets/base_nlp_dataset.py (hunk 2 above)
Comment (KPatr1ck) [reply paddlehub/datasets/base_nlp_dataset.py]: 已改

File: paddlehub/datasets/base_nlp_dataset.py (hunk 3 above)
Comment (KPatr1ck) [reply paddlehub/datasets/base_nlp_dataset.py]: 已改

File: paddlehub/module/nlp_module.py (hunk 5 above)
Comment (KPatr1ck) [reply paddlehub/module/nlp_module.py]: 已改

File: paddlehub/module/nlp_module.py (hunk 6 above)
Comment (KPatr1ck) [reply paddlehub/module/nlp_module.py:430]: 已改

File: modules/text/language_model/ernie_tiny/module.py (hunk 7 above)
Comment (KPatr1ck) [reply modules/text/language_model/ernie_tiny/module.py]: 已按review意见增加参数，原参数保持不变

File: paddlehub/module/nlp_module.py (hunk 4 above)
Comment (KPatr1ck) [reply paddlehub/module/nlp_module.py:410]: 因为text是一个列表，可能存在2个文本，为了方便理解而改动的
考虑到是内部使用的函数，所以将参数名从text改成了texts

File: demo/text_matching/predict.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+if __name__ == '__main__':
+    data = [
+        ['这个表情叫什么', '这个猫的表情叫什么'],
+        ['什么是智能手环', '智能手环有什么用'],
+        ['介绍几本好看的都市异能小说，要完结的！', '求一本好看点的都市异能小说，要完结的'],
+        ['一只蜜蜂落在日历上（打一成语）', '一只蜜蜂停在日历上（猜一成语）'],
+        ['一盒香烟不拆开能存放多久？', '一条没拆封的香烟能存放多久。'],
+    ]
+    label_map = {0: 'unmatch', 1: 'match'}
```
Comment (Steffy-zxf) [inline demo/text_matching/predict.py]: `{0: 'unmatched', 1: 'matched'}`

or

`{0: 'similar', 1: 'dissimilar'}`

File: paddlehub/datasets/lcqmc.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
+#coding:utf-8
```
Comment (Steffy-zxf) [inline paddlehub/datasets/lcqmc.py]: 可去掉（paddlehub 要求py3）

File: demo/text_matching/predict.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (Steffy-zxf) [inline demo/text_matching/predict.py]: 2021

File: demo/text_matching/train.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (Steffy-zxf) [inline demo/text_matching/train.py]: 2021

File: demo/text_matching/predict.py (hunk 8 above)
Comment (KPatr1ck) [reply demo/text_matching/predict.py]: 已改为'similar'和'dissimilar'

File: demo/text_matching/predict.py (hunk 10 above)
Comment (KPatr1ck) [reply demo/text_matching/predict.py]: 已更正

File: demo/text_matching/train.py (hunk 11 above)
Comment (KPatr1ck) [reply demo/text_matching/train.py]: 已更正

File: paddlehub/datasets/lcqmc.py (hunk 9 above)
Comment (KPatr1ck) [reply paddlehub/datasets/lcqmc.py]: 已删除

**Review by Steffy-zxf — APPROVED** [review]
LGTM
